docker/Dockerfile.multiarch

```diff
@@ -0,0 +1,67 @@
+FROM --platform=$BUILDPLATFORM rust:1.67.0-alpine as builder
+
+# Install build dependencies
+RUN apk add --no-cache git openssl-dev libpq-dev musl-dev
+
+# Set the working directory to /app and copy the source code
+WORKDIR /app
+COPY . .
+
+# Set the target architecture (can be set using --build-arg), buildx set it automatically
+ARG TARGETARCH
+
+# This can be set to release using --build-arg
+ARG RUST_RELEASE_MODE="debug"
+
+# Prepare toolchain
+# Docker and Rust use different architecture naming schemas, so we have to convert them
+RUN case $TARGETARCH in \
+      arm64) RUSTARCH=aarch64 ;; \
+      amd64) RUSTARCH=x86_64 ;; \
+      *) echo "unsupported architecture: $TARGETARCH"; exit 3 ;; \
+    esac \
+ && echo "RUSTARCH=$RUSTARCH" >> .buildenv
+
+RUN \
+    # Check if $TARGETPLATFORM is different from $BUILDPLATFORM
+    if [ "$TARGETPLATFORM" != "$BUILDPLATFORM" ]; then \
+        # Install necessary packages for cross-compiling OpenSSL if target architecture is different from build architecture
+        apk add --no-cache perl make gcc clang && \
+        # Set the FEATUREFLAG environment variable with the specified value
+        echo "FEATUREFLAG='--features vendored_openssl'" >> .buildenv; \
+        # fix https://github.com/briansmith/ring/issues/1414
+        echo "export CC_aarch64_unknown_linux_musl=clang" >> .buildenv; \
+        echo "export AR_aarch64_unknown_linux_musl=llvm-ar" >> .buildenv; \
```

Comment:
I take it these are from here: https://github.com/briansmith/ring/issues/1414#issuecomment-1055177218

It says you also might need to install a few dependencies too, which might be in the alpine repos.

Reply:
yes that is done in L29 ...

Reply:
uh you are right only llvm-libs is covered by dependency I have to add llvm too